pandas/tests/reshape/merge/test_merge.py

```diff
@@ -134,6 +134,18 @@ def test_merge_common(self):
         exp = merge(self.df, self.df2, on=["key1", "key2"])
         tm.assert_frame_equal(joined, exp)
 
+    def test_merge_non_string_columns(self):
+        # https://github.com/pandas-dev/pandas/issues/17962
+        # Checks that method runs for non string column names
+        left = pd.DataFrame(
+            {0: [1, 0, 1, 0], 1: [0, 1, 0, 0], 2: [0, 0, 2, 0], 3: [1, 0, 0, 3]}
+        )
+
+        right = left.astype(float)
+        expected = left
+
+        tm.assert_frame_equal(expected, pd.merge(left, right))
```

Comment:
Could you assign `result = pd.merge(left, right)` before this statement so this line can be `tm.assert_frame_equal(expected, result)

Reply:
this is now done